server/src/lib/model/leaderboard.ts

```diff
@@ -97,6 +138,122 @@ async function getVoteLeaderboard(locale?: string): Promise<any[]> {
   return rows;
 }
 
+// NOTE: The 4 top-related SQLs
+// 1) think the huge SQL as a process of adding aggregated attributes to user_clients rows:
+// first add bonus, then add clips as well as the votes for the clips, at last do calculation.
+// 2) only calculate values within the range of the challenge time span for bonus and clips, (maybe not votes).
+// 3) be careful in deciding where to put the filtering conditions: in LEFT JOIN or in WHERE, big differences.
+// 4) surely it can be optimized in that:
+//        participant in sql are ALMOST identical;
+//        topContributor and topMember sqls are different only in the WHERE clause.
+// 5) some of the conditions are probably over-thinking.
```

Comment:
Thanks for thinking about optimization. I'm happy to start testing unoptomized and see where we need attention.